### PR TITLE
Port Mesa wrapper script from Autopatcher, patch openssl

### PR DESCRIPTION
**Pull request by peeley**
This PR incorporates the changes from https://github.com/AM2R-Community-Developers/AM2R-Autopatcher-Linux/pull/35 into the launcher. Similarly to the `patcher.sh` script in that repo, these changes cause the launcher to do the following:
1. Patch the deprecated OpenSSL 1.0.0 dependency to point to `libcurl`
2. Wraps the `runner` binary with a script that prevents a Mesa bug from crashing the game on startup on AMD GPUs.

As discussed in the PR for `AM2R-Autopatcher-Linux`, the first item is gated behind a `PATCHOPENSSL` compile-time constant, and shouldn't affect the majority of users installing and launching the game via AppImage.

### CODE REVIEW

File: AM2RLauncher/AM2RLauncherLib/Profile.cs [hunk 1]
```diff
@@ -481,7 +481,28 @@ public static void InstallProfile(ProfileXML profile, bool useHqMusic, IProgress
...
+            patchLibcryptoProcess.StartInfo = patchLibcryptoStartInfo;
+            patchLibcryptoProcess.Start();
+            patchLibcryptoProcess.WaitForExit();
+
+            log.Info("Patched openssl.");
+
+            #endif
+
+            File.Move($"{tempPath}/{exe}", $"{tempPath}/.runner-unwrapped");
+            File.Move($"{Core.PatchDataPath}/data/mesa-wrapper.sh", $"{tempPath}/runner");
```
Comment (Miepee) [inline AM2RLauncher/AM2RLauncherLib/Profile.cs:505]: Pretty sure the last one is supposed to be a copy operation, not a moving operation.

**Review by Miepee — COMMENTED** [review]
Can you also add a log in AM2RLauncher.Gtk/Program.cs for the compile option?

**Comment by Miepee** [comment]
Don't mind the macos fail, apparently our setup/CI just needs reworking, completely unrelated to this PR.
Looks good now, thanks! 
If you need more help with the nix package, please do let me know! I really should also make a new release soon, but have just... kinda been putting it off, as I needed some more preperations on the linux/flatpak side.

**Comment by peeley** [comment]
No worries at all, thank you!

I'll keep an eye on the project for future releases too. My gamedev skills are close to zero, but I'm happy to help out with Linux/MacOS packaging and build stuff in the future.

**Comment by Miepee** [comment]
Oh hey that reminds me.
Dunno how big the mac userbase is on nix, but would there be any interest in packaging am2rlauncher.mac for the mac side of nix? Mac's still a real afterthought, but maybe that's just the push needed to give it a bit of cleanup to make it work properly.

**Comment by peeley** [comment]
The MacOS userbase on Nix definitely exists, but it's not nearly as popular or "blessed" as the Linux userbase.

I can definitely take a crack at getting the launcher to run on it, but currently the only device I'd be able to test on is an M1 Macbook Pro. I haven't even tried to run AM2R on `aarch64`, do you know if others have had any success?

**Comment by Miepee** [comment]
It does run and mostly works, assuming `xdelta3` is installed. Last time i tried it with someone else (I forgot which architecture it was tho...), it just "needed" another patch that adds the common Homebrew paths to PATH (which was also on my todo list before next release), because most people install xdelta that way, and due to non-login shell weirdness, they aren't in PATH if you open the launcher via double clicking instead of launching it from terminal.
AM2R itself tho should run on both x86(_64) and arm via rosetta.

**Pull request merged**